---
title: " Fast and accurate short-read alignment 
with hybrid hash-tree data structure"

authors:
- Junichiro Makino
- Toshikazu Ebisuzaki
- Ryutaro Himeno
- Yoshihide Hayashizaki
author_notes:
- "Equal contribution"
- "Equal contribution"
date: "2024-10-29"
doi: ""

# Schedule page publish date (NOT publication's date).
publishDate: "2024-10-29T00:00:00Z"

# Publication type.
# Accepts a single type but formatted as a YAML list (for Hugo requirements).
# Enter a publication type from the CSL standard.
publication_types: ["article-journal"]

# Publication name and optional abbreviated publication name.
publication: "Genomics & Informatics"
publication_short: ""

abstract: Rapidly increasing the amount of short-read data generated by NGSs (new-generation sequencers) calls for the development of fast and accurate read alignment programs. The programs based on the hash table (BLAST) and Burrows-Wheeler transform (bwa-mem) are used, and the latter is known to give superior performance. We here present a new algorithm, a hybrid of hash table and suffix tree, which we designed to speed up the alignment of short reads against large reference sequences such as the human genome. The total turnaround time for processing one human genome sample (read depth of 30) is just 31 min with our system while that was more than 25 h with bwa-mem/gatk. The time for the aligner alone is 28 min for our system but around 2 h for bwa-mem. Our new algorithm is 4.4 times faster than bwa-mem while achieving similar accuracy. Variant calling and other downstream analyses after the alignment can be done with open-source tools such as SAMtools and Genome Analysis Toolkit (gatk) packages, as well as our own fast variant caller, which is well parallelized and much faster than gatk.

# Summary. An optional shortened abstract.
summary: The artical introduces a new algorithm that combines hash table and suffix tree techniques to significantly speed up the alignment of short-read NGS data against large reference genomes like the human genome. This algorithm reduces the processing time of a human genome sample from over 25 hours to just 31 minutes, achieving a 4.4-fold increase in speed while maintaining comparable accuracy to traditional methods. It also supports fast, efficient variant calling with existing open-source tools and a proprietary variant caller.
tags:
- Human whole genome analysis
- Short read
- Alignment (mapping)
- Variant calling
- Hash
- Tree
featured: true

# links:
# - name: ""
#   url: ""
url_pdf: 'https://link.springer.com/content/pdf/10.1186/s44342-024-00012-5.pdf'
url_code: ''
url_dataset: ''
url_poster: ''
url_project: ''
url_slides: ''
url_source: ''
url_video: ''

# Featured image
# To use, add an image named `featured.jpg/png` to your page's folder. 
image:
  # caption: 'Image credit: [**Unsplash**](https://unsplash.com/photos/jdD8gXaTZsc)'
  focal_point: ""
  preview_only: false

# Associated Projects (optional).
#   Associate this publication with one or more of your projects.
#   Simply enter your project's folder or file name without extension.
#   E.g. `internal-project` references `content/project/internal-project/index.md`.
#   Otherwise, set `projects: []`.
projects: []

# Slides (optional).
#   Associate this publication with Markdown slides.
#   Simply enter your slide deck's filename without extension.
#   E.g. `slides: "example"` references `content/slides/example/index.md`.
#   Otherwise, set `slides: ""`.
slides: example
---
<!-- 
{{% callout note %}}
Click the *Cite* button above to demo the feature to enable visitors to import publication metadata into their reference management software.
{{% /callout %}}

{{% callout note %}}
Create your slides in Markdown - click the *Slides* button to check out the example.
{{% /callout %}} -->

<!-- Add the publication's **full text** or **supplementary notes** here. You can use rich formatting such as including [code, math, and images](https://docs.hugoblox.com/content/writing-markdown-latex/). -->
